cli/types/index.d.ts

```diff
@@ -174,757 +182,739 @@ declare namespace Cypress {
+     *
      * @see https://on.cypress.io/clearlocalstorage
      */
-    clearLocalStorage(key?: string): Chainable
-    clearLocalStorage(re: RegExp): Chainable
+    clearLocalStorage(key?: string): Chainable<null>
+    clearLocalStorage(re: RegExp): Chainable<null>
 
     /**
+     * Click a DOM element.
+     *
      * @see https://on.cypress.io/click
      */
-    click(options?: ClickOptions): Chainable
-    click(position: string, options?: ClickOptions): Chainable
-    click(x: number, y: number, options?: ClickOptions): Chainable
+    click(options?: Partial<ClickOptions>): Chainable<Subject>
+    click(position: string, options?: Partial<ClickOptions>): Chainable<Subject>
+    click(x: number, y: number, options?: Partial<ClickOptions>): Chainable<Subject>
+
+    /**
+     * `cy.clock()` overrides native global functions related to time allowing them to be controlled
+     * synchronously via [cy.tick()](https://on.cypress.io/tick) or the yielded clock object.
+     * This includes controlling:
+     * * `setTimeout`
+     * * `clearTimeout`
+     * * `setInterval`
+     * * `clearInterval`
+     * * `Date` Objects
+     *
+     * The clock starts at the unix epoch (timestamp of 0). This means that when you instantiate new Date in your application, it will have a time of January 1st, 1970.
+     *
+     * @see https://on.cypress.io/clock
+     */
+    clock(): Chainable<Clock>
+    clock(now: number, options?: Loggable): Chainable<Clock>
+    clock(now: number, functions?: Array<'setTimeout' | 'clearTimeout' | 'setInterval' | 'clearInterval'>, options?: Loggable): Chainable<Clock>
+    clock(options: Loggable): Chainable<Clock>
 
     /**
+     * Get the first DOM element that matches the selector (whether it be itself or one of its ancestors).
+     *
      * @see https://on.cypress.io/closest
      */
-    closest(selector: string, options?: LoggableTimeoutable): Chainable
+    closest<K extends keyof HTMLElementTagNameMap>(selector: K, options?: Partial<Loggable & Timeoutable>): Chainable<JQuery<HTMLElementTagNameMap[K]>>    
+    closest<E extends Node = HTMLElement>(selector: string, options?: Partial<Loggable & Timeoutable>): Chainable<JQuery<E>>
 
     /**
+     * Get the DOM element containing the text. DOM elements can contain more than the desired text and still match. Additionally, Cypress prefers some DOM elements over the deepest element found.
+     *
      * @see https://on.cypress.io/contains
      */
-    contains(text: string, options?: LoggableTimeoutable): Chainable
-    contains(num: number | RegExp): Chainable
-    contains(selector: string, text: string, options?: LoggableTimeoutable): Chainable
+    contains(content: string | number | RegExp): Chainable<Subject>
+    contains<E extends Node = HTMLElement>(content: string | number | RegExp): Chainable<JQuery<E>>

```

Comment:
These generic overrides are for when the user knows what the element is going to be and needs access to special properties or methods on the HTML element. It is probably going to be pretty rare, but may be required in rare circumstances (maybe a hacked file upload that requires an `HTMLInputElement` type.

The fallback for all generic overrides is `HTMLElement` which contains all the generic methods and properties. Also the `Subject` returned by all Element selectors is a `JQuery` object which helps to normalize a lot of calls.